state/src/item/metadata.rs

```diff
@@ -159,20 +177,27 @@ impl Encodable for Metadata {
             }
             s.append(&self.seq).append(self.params.as_ref().unwrap());
         }
+        if term_params_changed {
+            if !params_changed {
+                unreachable!("Term params only can be changed if params changed");
+            }
+            s.append(self.term_params.as_ref().unwrap());
+        }
     }
 }
 
 impl Decodable for Metadata {
     fn decode(rlp: &Rlp) -> Result<Self, DecoderError> {
-        let (term, seq, params) = match rlp.item_count()? {
-            4 => (TermMetadata::default(), 0, None),
+        let (term, seq, params, term_params) = match rlp.item_count()? {
+            4 => (TermMetadata::default(), 0, None, None),
```

Comment:
How about using constants like `INITIAL_LEN` defined above?
I think using hardcoded numbers is a bit hard to understand.